lib/semian/circuit_breaker.rb

```diff
@@ -1,13 +1,37 @@
 module Semian
   class CircuitBreaker
-    attr_reader :state
 
-    def initialize(exceptions:, success_threshold:, error_threshold:, error_timeout:)
+    def initialize(name, exceptions:, success_threshold:, error_threshold:, error_timeout:, permissions: 0660)
+      @name = "#{name}"
       @success_count_threshold = success_threshold
       @error_count_threshold = error_threshold
       @error_timeout = error_timeout
       @exceptions = exceptions
-      reset
+
+      @sliding_window = ::Semian::SysVSlidingWindow.new(
+        "#{name}_sliding_window",
+        @error_count_threshold,
+        permissions)
+      @successes = ::Semian::SysVAtomicInteger.new(
+        "#{name}_atomic_integer",
+        permissions)
+      @state = ::Semian::SysVAtomicEnum.new(
+        "#{name}_atomic_enum",
+        permissions,
+        [:closed, :half_open, :open])
+      # We do not need to #reset here since initializing is handled like this:
+      # (0) if data is not shared, then it's zeroed already
+      # (1) if no one is attached to the memory, zero it
+      # (2) otherwise, keep the data
+    end
+
+    #Replacement for attr_reader :state
```

Comment:
shouldn't this comment be on the `def state` method? ... imho comments like this are useless.. it's outdated already before you even shipped it :)